### PR TITLE
imporve `var-naming`  - add upperCaseConst  option to allow UPPER_CASED constants #851

### PR DESCRIPTION
**Pull request by comdiv**
it's improvement for  #851 task

### CODE REVIEW

File: rule/var-naming.go [hunk 1]
```diff
@@ -82,25 +103,31 @@ func (*VarNamingRule) Name() string {
...
 	}
 	for _, f := range fl.List {
 		for _, id := range f.Names {
-			check(id, thing, w)
+			w.check(id, thing)
 		}
 	}
 }
 
-func check(id *ast.Ident, thing string, w *lintNames) {
```
Comment (comdiv) [inline rule/var-naming.go:96]: It was too gready to "reserve" name `check` in package rule just for var-naming, moved lintNames as reviever to minimize exported named

**Comment by chavacava** [comment]
Hi @comdiv, thanks for the PR! (and sorry for the late response)
I've made some little modifications to the code. 
Could you please provide a fix for the failing test case and update the description of the rule in RULES_DESCRIPTIONS.md ?

**Comment by comdiv** [comment]
> Hi @comdiv, thanks for the PR! (and sorry for the late response) I've made some little modifications to the code. Could you please provide a fix for the failing test case and update the description of the rule in RULES_DESCRIPTIONS.md ?

Just not understand failing tests for now. Locally all work well.... Will try to fix

**Comment by chavacava** [comment]
> Just not understand failing tests for now. Locally all work well.... Will try to fix

Test fails because I' ve added à test case for constant names without `_` (e.g. `VER`,  `MAX`, ...) 
You might need to pull locally the modifications I've made on your branch.

**Comment by comdiv** [comment]
Ok, pulled.
But it's not after my changes, in legacy code we see:

```
if len(id.Name) >= 5 && allCapsRE.MatchString(id.Name) && strings.Contains(id.Name, "_") 
```
so your example of `VER` failed in old version of rule too.

What is better:

1. Change your VER example to VERSION one
2. Remove len limit for uppercase totally
3. Move len limit to parameter
4. Make len 1 or 2?

**Comment by chavacava** [comment]
I see. The >= 5 condition comes from `golint` code base.
Please remove the test case I' ve added and keep te condition as is.
Thanks

**Comment by comdiv** [comment]
Fix MD file too.

**Comment by chavacava** [comment]
Thansk @comdiv !

**Pull request merged**